website/content/docs/templates/hcl_templates/blocks/build/provisioner.mdx

```diff
@@ -76,6 +76,39 @@ build {
 
 The values within `only` or `except` are _build names_, not builder types.
 
+## On Error Provisioner
+
+You can optionally create a single specialized provisioner called an
+`error-cleanup-provisioner`. This provisioner will not run unless the normal
+provisioning run fails. If the normal provisioning run does fail, this special
+error provisioner will run _before the instance is shut down_. This allows you
+to make last minute changes and clean up behaviors that Packer may not be able
+to clean up on its own.
+
+For examples, users may use this provisioner to make sure that the instance is
+properly unsubscribed from any services that it connected to during the build
+run.
+
+Toy usage example for the error cleanup script:
+
+```hcl
+source "null" "example" {
+  communicator = "none"
+}
+
+build {
+  sources = ["source.null.example"]
+
+  provisioner "shell-local" {
+    inline = ["exit 2"]
+  }
+
+  error-cleanup-provisioner "shell-local" {
+    inline = ["echo 'rubber ducky'> ducky.txt"]
+  }
```

Comment:
Here the syntax is super similar to the one of a `provisioner`; and except for the docs or a runtime error, nothing says that I can only add only one of these here. Is there a way to get that ? Should we allow more than one ?

Reply:
I think we should allow only one for now and keep the current behavior. Otherwise, we will have to change more code.

Reply:
Or if we decide to add that possibility I would rather do it in another PR as an enhancement

Reply:
Sounds good !